### ISSUE TITLE
Swimlane Rules don't work

### ISSUE THREAD

**Issue by vkleinmp** [wekan/wekan#2225]
Found bugs with rules regarding swimlanes like:

if card is moved to list * in swinlane REVIEW by *
set color to yellow

by error apply to all swimlanes and every card being move to ANY swimlane is set to yellow.

Same problems if setting the list AND swimlane or using "moves from list *" instead of "moved to list*"
using Wekan 2.27 as Docker-container

**Comment by xet7** [comment]
Huh? I'm currently trying to fix current Wekan v2.37. Old versions are not supported.

**Comment by xet7** [comment]
Try Wekan v2.35 is the problem there.

**Comment by vkleinmp** [comment]
2.37 still has the problems. 
rules of the type "if card is moved to list * in swinlane SWNAME by *" are applied on ALL card on moving, regardless of swimlane, even in "Default".

**Comment by vkleinmp** [comment]
Any chance you could work on this soon? Bug is breaking our projects right now.

**Comment by xet7** [comment]
@vkleinmp 

Well, if someone adds bounty or contacts my Commercial Support as described at https://wekan.team , then Yes. Otherwise, it's sometime after all other work.

**Comment by xet7** [comment]
Hmm, I could try to figure it out, no promises though yet. In general, I do plan to fix all bugs.

@Angtrim did originally develop Rules feature, so it would take less time for him to fix it, if he has free time, than me to get up-to-speed.

**Comment by Angtrim** [comment]
@vkleinmp I've already identified the bug. 
It was a problem of collecting the swimlane name from the right field.
Right now the swimlane name is (wrongfully) collected from the "When a card is ADDED..." swimlane name field.
Sot if it is critical, a possible workaround is to write the swimlane name there when using the "MOVED TO" rule.

@xet7 My fix is included in a commit on the "cloning feature" branch I still have to PR.
Should I separate it in a different branch? Or it's ok if I PR everything?

**Comment by xet7** [comment]
@Angtrim

It would be safer in separate branch and PR.

**Comment by xet7** [comment]
@Angtrim 

Actually, I'll try your branch, does it work.

**Comment by xet7** [comment]
@Angtim 

To fix lint errors, I did make following change:
```
diff --git a/models/export.js b/models/export.js
index d402efe..49aac82 100644
--- a/models/export.js
+++ b/models/export.js
@@ -30,7 +30,7 @@ if (Meteor.isServer) {
     }
 
     const exporter = new Exporter(boardId);
-    if (true||exporter.canExport(user)) {
+    if (exporter.canExport(user)) {
       JsonRoutes.sendResult(res, {
         code: 200,
         data: exporter.build(),
```
Maybe you added that `true` just for testing?

**Comment by Angtrim** [comment]
@xet7  Yes, nice catch. It's probably from when I was still trying to figure out what was blocking the export.

**Comment by xet7** [comment]
Fixed in Wekan v2.58 and newer.
